easybuild/easyblocks/c/cp2k.py

```diff
@@ -272,14 +270,32 @@ def configure_common(self):
             regflags = 'NOOPT'
 
         # make sure a MPI-2 able MPI lib is used
-        mpi2libs = ['impi', 'MVAPICH2', 'OpenMPI']
         mpi2 = False
-        for mpi2lib in mpi2libs:
-            if get_software_root(mpi2lib):
+        if hasattr(self.toolchain, 'MPI_FAMILY') and self.toolchain.MPI_FAMILY is not None:
+            mpi_spec_by_fam = {
+                toolchain.MPICH: 'mpi2',  # MPICH is MPICH v3.x, which is MPI2 compatible
+                toolchain.MPICH2: 'mpi2',
+                toolchain.MVAPICH2: 'mpi2',
+                toolchain.OPENMPI: 'mpi2',
+                toolchain.IMPI: 'mpi2',
+            }
+            mpi_fam = self.toolchain.mpi_family()
+            mpi_spec = mpi_spec_by_fam.get(mpi_fam)
```

Comment:
since the value for `mpi_spec` is the same regardless of which toolchain it is, this can be cleaned up a bit to:

```python
known_mpi2_fams = [toolchain.MPICH, toolchain.MPICH2, toolchain.MVAPICH2, toolchan.OPENMPI, toolchain.IMPI]
if self.toolchain.mpi_family() in known_mpi2_fams:
    mpi2 = True
```